pkg/util/informermanager/federatedinformermanager.go

```diff
@@ -0,0 +1,328 @@
+package informermanager
+
+import (
+	"bytes"
+	"context"
+	"encoding/gob"
+	"fmt"
+	"sync"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/dynamic"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog/v2"
+
+	fedcorev1a1 "github.com/kubewharf/kubeadmiral/pkg/apis/core/v1alpha1"
+	fedcorev1a1informers "github.com/kubewharf/kubeadmiral/pkg/client/informers/externalversions/core/v1alpha1"
+	fedcorev1a1listers "github.com/kubewharf/kubeadmiral/pkg/client/listers/core/v1alpha1"
+	"github.com/kubewharf/kubeadmiral/pkg/controllers/util"
+)
+
+type federatedInformerManager struct {
+	lock sync.RWMutex
+
+	started bool
+
+	clientGetter    ClusterClientGetter
+	ftcInformer     fedcorev1a1informers.FederatedTypeConfigInformer
+	clusterInformer fedcorev1a1informers.FederatedClusterInformer
+
+	eventHandlerGenerators []*EventHandlerGenerator
+	clusterEventHandler    []*ClusterEventHandler
+
+	clients                     map[string]dynamic.Interface
```

Comment:
I'm wondering if returning the underlying transport would make this easier to use. With the transport, users should be able to construct their own clients, be it typed or untyped.

Reply:
We could, but the purpose of `InformerManager` and `FederatedInformerManager` is to provide an abstraction to work with the FTC source types, which is only known at runtime and users would be forced to use a dynamic client anyway.